Make cve module self contained and not require stdlib

jeffmccune opened the pull request:
Without this patch, the cve20113872 module made use of the validate_re()
function from stdlib.  To minimize the number of steps the end user
needs to make this patch copies the function into this module.

This means stdlib is no longer required and the function has been
renamed to prevent conflicts with stdlib should it already be installed
which is the case on PE 1.2.3.

In addition, the README, module metadata and Rakefile have been updated
to reflect this change.  The rake tasks have also been modified to
function if rspec is not installed, which is the case on a PE system.

Finally, a build and clean task have been added to help clean out and
rebuild the cve module packages.

Commit:
Make cve module self contained and not require stdlib

Without this patch, the cve20113872 module made use of the validate_re()
function from stdlib.  To minimize the number of steps the end user
needs to make this patch copies the function into this module.

This means stdlib is no longer required and the function has been
renamed to prevent conflicts with stdlib should it already be installed
which is the case on PE 1.2.3.

Closes #7

jhelwig commented:
Looking at this one.

jhelwig commented on Modulefile:
Seems like everything from here up should be in a separate commit?

jhelwig commented:
I'm a bit confused by the changes to `bin/pe_step1_secure_the_master` since they don't appear to be discussed anywhere in the commit message, also I'm not sure how the README changes are related to the stdlib change.

jeffmccune commented on Modulefile:
Agreed, I'll pull it out into its own.

jeffmccune commented:
Yeah, I modified the step1 script because it was stoping the service before it checked if there was a backup already.  This meant the service would stay down if we bail out.

I'll ammend the commit message.

Commit:
Update module metadata URL's

The module metadata was generated when we didn't have the Github
repository setup yet.  Without this patch the metadata is stale and out
of date.

This patch updates the Module metadata to bring it up to speed with the
current "main" location of the code.

Commit:
Update README with module build information

The README file is out of date without this patch applied.  The code has
since moved to a Puppet Module and we are assuming the first thing the
end user will do is install it into their module path.

Commit:
Add build and clean tasks to the Rakefile

Without this patch, the only rake tasks available are testing related.
Even worse, rake -T doesn't work with stock PE since the Rakefile tries
to load rspec.

This patch fixes this issue by making the rspec tests optional and
unavailable if rspec isn't loadable.

In addition, two simple tasks, build and clean, are added by this patch
which build the puppet module and clean out the package directory
respectively.

Commit:
Do not leave the master stopped

Without this patch, if the user runs through step1 twice, then we bail
out when we detect the backup copy.  We also leave their master shut
down, which isn't very nice.

This patch moves the chunk of code that stops the service after the
chunk of code that bails out if the backup file is present.

This results in us leaving their services the way they are if we refuse
to run a second time.

jeffmccune commented:
@jhelwig I cleaned up this topic branch.  Please re-review and merge if it's good now.
